### PR TITLE
Update to SmallRye-GraphQL 2.2.0

### PR DESCRIPTION
**Pull request by mskacelik**
/cc @jmartisk

### CODE REVIEW

**Comment by jmartisk** [comment]
@brunobat / @radcortez FYI this replaces OpenTracing support in GraphQL with OpenTelemetry. We've identified some weird behavior with more complex GraphQL operations though, so for now we're keeping the tests basic. Need to look into that later.

File: extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/instrumentation/GraphQLOpenTelemetryTest.java [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+                        Collectors.toList()))
+                .values().stream()
+                .collect(Collectors.toList());
+
+        assertEquals(spanCollections.size(), iterations);
+        for (List<SpanData> spanGroup : spanCollections) {
+            assertTimeForSpans(spanGroup);
+            assertHttpSpan(spanGroup);
+
+            final SpanData operationSpan = spanGroup.get(2);
```
Comment (brunobat) [inline extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/instrumentation/GraphQLOpenTelemetryTest.java:164]: Using order can be tricky and lead to transient failures.
Using `getSpanByKindAndParentId(...)` is recomended.

**Review by brunobat — COMMENTED** [review]
You should probably add some documentation to the OpenTelemetry documentation page.
Also I don't see anything in relation to the graphQL client... 
What were the complex scenarios that were failing?

**Review by phillip-kruger — APPROVED** [review]

**Comment by jmartisk** [comment]
>  Also I don't see anything in relation to the graphQL client... 

We don't support GraphQL client yet, only server. In the future we want to propagate the tracing context from client to server. For now, the whole support is just generating a span for the GraphQL operation as a whole.

> What were the complex scenarios that were failing?´

Uhmm, if I remember correctly, anything where the GraphQL operation involved `@Source` methods caused some unpredictable random issues with tracing context propagation and scope closing. And there were some more that I don't recall right now. We've dropped it for now but want to come back to it later.

**Review by jmartisk — APPROVED** [review]

**Comment by rsvoboda** [comment]
https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.1#smallrye-graphql was added

**Pull request merged**